src/schema/sources.ts

```diff
@@ -865,6 +891,36 @@ export const resolvers: IResolvers<any, Context> = {
       });
       return { _: true };
     },
+    updateMemberRole: async (
+      _,
+      { sourceId, memberId, role }: UpdateMemberRoleArgs,
+      ctx,
+    ): Promise<GQLEmptyResponse> => {
+      if (role === SourceMemberRoles.Blocked) {
```

Comment:
We also have to think about "unblocking" basically it would be when assigning the "Member" role again, so maybe use `.includes`?